In [5]:
import pandas as pd 
import numpy as np

# Carregando o dataset de pagamentos
pgtos_df = pd.read_csv("olist_order_payments_dataset.csv")
pgtos_df



order_id  payment_sequential payment_type  \
0       b81ef226f3fe1789b1e8b2acac839d17                   1  credit_card   
1       a9810da82917af2d9aefd1278f1dcfa0                   1  credit_card   
2       25e8ea4e93396b6fa0d3dd708e76c1bd                   1  credit_card   
3       ba78997921bbcdc1373bb41e913ab953                   1  credit_card   
4       42fdf880ba16b47b59251dd489d4441a                   1  credit_card   
...                                  ...                 ...          ...   
103881  0406037ad97740d563a178ecc7a2075c                   1       boleto   
103882  7b905861d7c825891d6347454ea7863f                   1  credit_card   
103883  32609bbb3dd69b3c066a6860554a77bf                   1  credit_card   
103884  b8b61059626efa996a60be9bb9320e10                   1  credit_card   
103885  28bbae6599b09d39ca406b747b6632b1                   1       boleto   

        payment_installments  payment_value  
0                          8          99.33  
1                          1          24.39  
2                          1          65.71  
3                          8         107.78  
4                          2         128.45  
...                      ...            ...  
103881                     1         363.31  
103882                     2          96.80  
103883                     1          47.77  
103884                     5         369.54  
103885                     1         191.58  

[103886 rows x 5 columns]

In [49]:
# Checando se há valores nulos no dataset
print(pgtos_df.isnull().sum())

order_id                0
payment_sequential      0
payment_type            0
payment_installments    0
payment_value           0
dtype: int64


In [9]:
# Cabeçalho do dataframe
pgtos_df.head()

order_id  payment_sequential payment_type  \
0  b81ef226f3fe1789b1e8b2acac839d17                   1  credit_card   
1  a9810da82917af2d9aefd1278f1dcfa0                   1  credit_card   
2  25e8ea4e93396b6fa0d3dd708e76c1bd                   1  credit_card   
3  ba78997921bbcdc1373bb41e913ab953                   1  credit_card   
4  42fdf880ba16b47b59251dd489d4441a                   1  credit_card   

   payment_installments  payment_value  
0                     8          99.33  
1                     1          24.39  
2                     1          65.71  
3                     8         107.78  
4                     2         128.45

In [14]:
# Formato do dataframe (linhas e colunas)
print(f'Linhas: {pgtos_df.shape[0]}\nColunas: {pgtos_df.shape[1]}')


Linhas: 103886
Colunas: 5


In [15]:
# Informações gerais do dataframe
pgtos_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 103886 entries, 0 to 103885
Data columns (total 5 columns):
 #   Column                Non-Null Count   Dtype  
---  ------                --------------   -----  
 0   order_id              103886 non-null  object 
 1   payment_sequential    103886 non-null  int64  
 2   payment_type          103886 non-null  object 
 3   payment_installments  103886 non-null  int64  
 4   payment_value         103886 non-null  float64
dtypes: float64(1), int64(2), object(2)
memory usage: 4.0+ MB


In [19]:
#Verificando quantos tipos de pagamentos existem e qual o mais utilizado
tipos_pgtos = pgtos_df['payment_type']

tipos_pgtos.describe()

count          103886
unique              5
top       credit_card
freq            76795
Name: payment_type, dtype: object

O tipo de pagamento mais utilizado é com cartão de crédito, ocorrendo 76795 vezes.
Há 5 tipos diferentes de pagamentos.

In [20]:
#Analisando as ocorrências por tipo de pagamento
tipos_pgtos.value_counts()

payment_type
credit_card    76795
boleto         19784
voucher         5775
debit_card      1529
not_defined        3
Name: count, dtype: int64

In [25]:
# Analisando o parcelamento das compras
parcelas = pgtos_df['payment_installments']
parcelas.describe()

count    103886.000000
mean          2.853349
std           2.687051
min           0.000000
25%           1.000000
50%           1.000000
75%           4.000000
max          24.000000
Name: payment_installments, dtype: float64

A média de parcelas das compras é 2,85.
O parcelamento máximo para este DF é 24 parcelas.

In [26]:
parcelas.value_counts() # Verificando a ocorrência de parcelamento das compras, ordenado por frequência.

payment_installments
1     52546
2     12413
3     10461
4      7098
10     5328
5      5239
8      4268
6      3920
7      1626
9       644
12      133
15       74
18       27
11       23
24       18
20       17
13       16
14       15
17        8
16        5
21        3
0         2
22        1
23        1
Name: count, dtype: int64

In [44]:
# Analisando quantas ocorrências de parcelamentos, ordenado em ordem crescente para o número de parcelas
sort_parcelas = pgtos_df.sort_values(by = 'payment_installments')
sort_parcelas
contagem_parcelas = sort_parcelas['payment_installments'].value_counts()
print(contagem_parcelas.sort_index())


payment_installments
0         2
1     52546
2     12413
3     10461
4      7098
5      5239
6      3920
7      1626
8      4268
9       644
10     5328
11       23
12      133
13       16
14       15
15       74
16        5
17        8
18       27
20       17
21        3
22        1
23        1
24       18
Name: count, dtype: int64


In [45]:
# Análise dos valores de pagamentos
valor_pgto = pgtos_df['payment_value']
valor_pgto

0          99.33
1          24.39
2          65.71
3         107.78
4         128.45
           ...  
103881    363.31
103882     96.80
103883     47.77
103884    369.54
103885    191.58
Name: payment_value, Length: 103886, dtype: float64

In [46]:
#Informações gerais
valor_pgto.describe()

count    103886.000000
mean        154.100380
std         217.494064
min           0.000000
25%          56.790000
50%         100.000000
75%         171.837500
max       13664.080000
Name: payment_value, dtype: float64

O valor médio de pagamento é 154,1.
O maior pagamento foi 13664.

In [52]:
# Verificando se há pagamentos do tipo 'boleto' que possuem mais de uma parcela:
pgtos_df[(pgtos_df['payment_installments']>1) & (pgtos_df['payment_type'] =="boleto")]

Empty DataFrame
Columns: [order_id, payment_sequential, payment_type, payment_installments, payment_value]
Index: []

In [63]:
#Qual a média de pagamentos
media_pgtos = pgtos_df['payment_value'].mean()
media_pgtos

154.10038041699553

In [64]:
#Quais são os pagamentos que têm um valor maior do que a média de pagamentos registrados?
pgtos_df[pgtos_df['payment_value']>media_pgtos]

order_id  payment_sequential payment_type  \
8       1f78449c87a54faf9e96e88ba1491fa9                   1  credit_card   
10      d88e0d5fa41661ce03cf6cf336527646                   1  credit_card   
15      12e5cfe0e4716b59afb0e0f4a3bd6570                   1  credit_card   
18      8ac09207f415d55acff302df7d6a895c                   1  credit_card   
21      4214cda550ece8ee66441f459dc33a8c                   1  credit_card   
...                                  ...                 ...          ...   
103879  c45067032fd84f4cf408730ff5205568                   1  credit_card   
103880  7159096c5aa9be77f7f0c26c01ee9793                   1  credit_card   
103881  0406037ad97740d563a178ecc7a2075c                   1       boleto   
103884  b8b61059626efa996a60be9bb9320e10                   1  credit_card   
103885  28bbae6599b09d39ca406b747b6632b1                   1       boleto   

        payment_installments  payment_value  
8                          6         341.09  
10                         8         188.73  
15                        10         157.45  
18                         4         244.15  
21                         2         170.57  
...                      ...            ...  
103879                     2         198.94  
103880                     4         280.65  
103881                     1         363.31  
103884                     5         369.54  
103885                     1         191.58  

[31012 rows x 5 columns]

In [65]:
#Quais são os pagamentos que têm um valor memnor do que a média de pagamentos registrados?
pgtos_df[pgtos_df['payment_value']<media_pgtos]

order_id  payment_sequential payment_type  \
0       b81ef226f3fe1789b1e8b2acac839d17                   1  credit_card   
1       a9810da82917af2d9aefd1278f1dcfa0                   1  credit_card   
2       25e8ea4e93396b6fa0d3dd708e76c1bd                   1  credit_card   
3       ba78997921bbcdc1373bb41e913ab953                   1  credit_card   
4       42fdf880ba16b47b59251dd489d4441a                   1  credit_card   
...                                  ...                 ...          ...   
103874  d41f1036712e18ee525939d08b7a1c98                   1  credit_card   
103876  0872d0faafe0cb56e6f2c594cca1522c                   1  credit_card   
103878  2e3d96522c8c378d4135c12a65c977aa                   1  credit_card   
103882  7b905861d7c825891d6347454ea7863f                   1  credit_card   
103883  32609bbb3dd69b3c066a6860554a77bf                   1  credit_card   

        payment_installments  payment_value  
0                          8          99.33  
1                          1          24.39  
2                          1          65.71  
3                          8         107.78  
4                          2         128.45  
...                      ...            ...  
103874                     1           3.57  
103876                     2         127.92  
103878                     4          47.62  
103882                     2          96.80  
103883                     1          47.77  

[72874 rows x 5 columns]

In [70]:
#Inserindo uma nova coluna booleana para pagamentos parcelados em pelo menos 12 vezes
pgtos_df['parcela_1_ano'] = pgtos_df['payment_installments'] >=12
pgtos_df['parcela_1_ano'].value_counts()


parcela_1_ano
False    103568
True        318
Name: count, dtype: int64

Existem 103568 pagamentos parcelados em menos de 12 vezes, e 318 pagamentos parcelados em mais de 12 vezes.